src/modules/github/components/BulkMergePrs.tsx

```diff
@@ -1,27 +1,101 @@
 'use client'
 
-import type { FC } from 'react'
+import { type FC, useEffect, useState } from 'react'
+
+import { DataList } from '@/components/github/root/DataList'
+import { PullRequestReviewForm } from '@/components/github/root/PullRequestReviewForm'
 
 import { usePrsGroup } from '@/hooks/github/root/usePrsGroup'
 
+import { type TRepoHasCheck } from '@/types/github/root/index'
+
 export const BulkMergePrs: FC = () => {
   const { prsGroup } = usePrsGroup()
+  const [repoHasCheckArray, setRepoHasCheckArray] = useState<TRepoHasCheck[]>(
+    []
+  )
+
+  const handlePullChange = (repo: string, pullNumber: number) => {
+    setRepoHasCheckArray((prev) =>
+      prev.map((data) => {
+        if (data.repo === repo) {
+          return {
+            ...data,
+            pulls: data.pulls.map((pull) => {
+              if (pull.number === pullNumber) {
+                return { ...pull, isChecked: !pull.isChecked }
+              }
+              return pull
+            })
+          }
+        }
+        return data
+      })
+    )
+  }
+
+  const handleRepoChange = (repo: string) => {
+    setRepoHasCheckArray((prev) =>
+      prev.map((data) => {
+        if (data.repo === repo) {
+          if (data.pulls.every((pull) => pull.isChecked === true)) {
```

Comment:
if `data.pulls` length is 0, this condition returns true. So you need additional condition.
```typescript
if (data.pulls.length > 0 && data.pulls.every((pull) => pull.isChecked === true)
```

Reply:
Ahh!!!! OKay

Reply:
Do I need to push?